### ISSUE TITLE
Group / permission editing is not working properly

### ISSUE THREAD

**Issue by DizzyOnion** [thibaud-rohmer/PhotoShow#276]
I recently installed PhotoShow on a small ARM powered server running Debian Squeeze (4.4.5) and PHP 5.3.3 and I'm facing an issue with the group editing / user permissions:
The main user's "root" privilege gets lost after klicking "submit" on the group editing page. Other users permissions (currently 2 in total) get messed up as well. I couldn't figure out what it could be.
The only way to re-enable admin privileges is to edit the accounts.xml directly. All other changes in settings seem to work properly so I doubt it is an issue with file read/write permissions.

**Comment by Heather705** [comment]
I had the same result.

**Comment by ltGuillaume** [comment]
Same here. Furthermore, after putting a user into the uploaders group manually via the xml, I tried to give the group uploaders rights to a certain private folder, but it does not allow them to upload there. I reckon this isn't by design? Or can uploaders only upload to public folders?

**Comment by stykoo** [comment]
Like @ltGuillaume I don't understand what the uploaders group is for, or how to give it the right to upload. With my settings only the root users can upload (in both public and private folders).

**Comment by nielsswinkels** [comment]
I can confirm this is still an issue. Tested with both firefox and chrome browser.

**Comment by gboudreau** [comment]
Can you describe in details what you do, and what happens?
I tried to reproduce what is detailed in the initial post, and can't find any problems:
- I started from the Groups editing page @ `/?t=Adm&a=EdA`, using the only admin account I have configured.
- Clicked Submit
- `Conf/accounts.xml` didn't change
- Tried to create a new group; was correctly added to `Conf/groups.xml`
- Added my account to the new group; `Conf/accounts.xml` was modified accordingly.
- I never lost root access at any point.

**Comment by nielsswinkels** [comment]
I had an external person who created an account. Then I created a group. Then I tried to add that person to the group by adding a checkmark for that person in that group, and hit submit. That instead removed the checkmark for my account being root, and didn't make any other changes. 

This is from how I remember what I've done. I can retry it tomorrow with a clean install and write down precise steps to reproduce it.

**Comment by gboudreau** [comment]
Might help. I tried that, and can't reproduce either:
- enabled registrations
- created a new account `bob`
- using admin account, when in Groups page; `bob` is part of the `user` group (default)
- Create a group `testGroup`
- Added `bob` to that group

**Comment by nielsswinkels** [comment]
Now I tried twice with a clean install and I can't reproduce the error there, but on my initial setup it still exists. Weird. I guess I will just reinstall it.

**Comment by nielsswinkels** [comment]
It seems that a capital in the username might cause the error.

**Comment by gboudreau** [comment]
Indeed, I was able to reproduce with `BobTest` as a username. Will debug with that. Thx
